rpc/src/responses/json_array.rs

```diff
@@ -1,13 +1,14 @@
 use super::ParseError;
-use serde::de;
+use serde::{de, Deserialize, Serialize};
 use serde_json::{self, json, Value};
 
+#[derive(Serialize, Deserialize)]
 pub struct JsonArray<T> {
     flattened_vec: Vec<T>,
 }
 
 impl<T: de::DeserializeOwned> JsonArray<T> {
-    pub fn from_str(str_to_parse: &str) -> Result<Self, ParseError> {
+    pub fn from_response(str_to_parse: &str) -> Result<Self, ParseError> {
```

Comment:
this feels a little disingenuous since we have `Response` structs, and we're taking in `str`s. 

Maybe to keep with the project's ubiquitous language, we should rename to `from_response_str`.

Reply:
yeah, that would 100% be better!

also LOL on the edit